If portable hotspot configuration fails, no feedback is given

On two of the devices I've tried (Galaxy Tab A and Alcatel One Touch 5020D, details at https://github.com/opendatakit/collect#contributing-testing), when I tap "Send" and select filled forms, I go straight to a screen that shows "Waiting for connection to accept". What does this state means? It looks like it's not possible to set up a hostpot on those devices? I don't get promtped to enable from the settings.

On the Moto G4 I do get prompted to enable the hotspot from settings. Then I see the QR code and port number information.

To summarize some new insights from Slack:

- Galaxy Tab A, Android 5.1.1: it seems I consistently get 
```
07-19 06:44:23.089 12413-12413/org.odk.share E/WifiHotspotHelper: java.lang.reflect.InvocationTargetException
        at java.lang.reflect.Method.invoke(Native Method)
        at java.lang.reflect.Method.invoke(Method.java:372)
        at org.odk.share.controller.WifiHotspotHelper.toggleHotspot(WifiHotspotHelper.java:139)
        at org.odk.share.controller.WifiHotspotHelper.enableHotspot(WifiHotspotHelper.java:129)
        at org.odk.share.activities.SendActivity.initiateHotspot(SendActivity.java:146)
        at org.odk.share.activities.SendActivity.startHotspot(SendActivity.java:130)
        at org.odk.share.activities.SendActivity.onCreate(SendActivity.java:105)
        at android.app.Activity.performCreate(Activity.java:6609)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1134)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3104)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3261)
        at android.app.ActivityThread.access$1000(ActivityThread.java:219)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1735)
        at android.os.Handler.dispatchMessage(Handler.java:102)
        at android.os.Looper.loop(Looper.java:145)
        at android.app.ActivityThread.main(ActivityThread.java:6939)
        at java.lang.reflect.Method.invoke(Native Method)
        at java.lang.reflect.Method.invoke(Method.java:372)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1404)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1199)
     Caused by: java.lang.SecurityException: ConnectivityService: Neither user 10229 nor current process has android.permission.CHANGE_NETWORK_STATE.
        at android.os.Parcel.readException(Parcel.java:1546)
        at android.os.Parcel.readException(Parcel.java:1499)
        at android.net.wifi.IWifiManager$Stub$Proxy.setWifiApEnabled(IWifiManager.java:1673)
        at android.net.wifi.WifiManager.setWifiApEnabled(WifiManager.java:2001)
```

- One Touch, Android 4.1.1:
No exception. The hotspot is set up and other devices do see it but then they prompt for a port number but the sending device doesn't show a port number.

--- a/share_app/src/main/AndroidManifest.xml
+++ b/share_app/src/main/AndroidManifest.xml
@@ -6,6 +6,7 @@
     <uses-permission android:name="android.permission.CHANGE_WIFI_STATE" />
     <uses-permission android:name="android.permission.ACCESS_WIFI_STATE" />
     <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
+    <uses-permission android:name="android.permission.CHANGE_NETWORK_STATE" />
     <uses-permission
         android:name="android.permission.WRITE_SETTINGS"
         tools:ignore="ProtectedPermissions" />

--- a/share_app/src/main/java/org/odk/share/activities/SendActivity.java
+++ b/share_app/src/main/java/org/odk/share/activities/SendActivity.java
@@ -68,6 +68,7 @@ public class SendActivity extends InjectableActivity {
     @BindView(R.id.tvConnectInfo)
     TextView connectInfo;
 
+    private boolean isHotspotInitiated;
     private boolean isHotspotRunning;
     private boolean openSettings;
     private ProgressDialog progressDialog;
@@ -96,13 +97,13 @@ public class SendActivity extends InjectableActivity {
             finish();
         }
 
+        isHotspotInitiated = false;
         isHotspotRunning = false;
         openSettings = false;
 
         if (wifiHotspot.isHotspotEnabled()) {
             wifiHotspot.disableHotspot();
         }
-        startHotspot();
     }
 
     /**
@@ -193,6 +194,11 @@ public class SendActivity extends InjectableActivity {
         super.onResume();
         compositeDisposable.add(addHotspotEventSubscription());
         compositeDisposable.add(addUploadEventSubscription());
+
+        if (!isHotspotInitiated) {
+            isHotspotInitiated = true;
+            startHotspot();
+        }
 
         if (openSettings) {
             openSettings = false;

--- a/share_app/src/main/java/org/odk/share/activities/WifiActivity.java
+++ b/share_app/src/main/java/org/odk/share/activities/WifiActivity.java
@@ -368,7 +368,9 @@ public class WifiActivity extends InjectableActivity {
                 NetworkInfo info = cm.getActiveNetworkInfo();
                 if (info != null) {
                     Timber.d(info + " " + info.getTypeName() + " " + info.getType() + " " + wifiNetworkSSID + " " + isConnected);
-                    if (info.getState() == NetworkInfo.State.CONNECTED && info.getTypeName().compareTo("WIFI") == 0) {
+                    if (info.getState() == NetworkInfo.State.CONNECTED
+                            && info.getTypeName().compareTo("WIFI") == 0
+                            && info.getExtraInfo() != null) {
                         if (!isConnected && info.getExtraInfo().equals("\"" + wifiNetworkSSID + "\"")) {
                             Timber.d("Connected");
                             isConnected = true;

